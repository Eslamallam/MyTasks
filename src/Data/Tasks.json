[
  {
    "id": 1,
    "category": "Sport",
    "description": "Run for 10 min"
  },
  {
    "id": 2,
    "category": "Entertainment",
    "description": "Watch a movie"
  },
  {
    "id": 3,
    "category": "Work",
    "description": "Do my tasks"
  }
]
