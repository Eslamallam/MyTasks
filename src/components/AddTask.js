import React from "react";
import { addNewTask } from "../actions/Task";
import TaskForm from "./TaskForm";
import { connect } from "react-redux";

const AddTask = props => {
  return (
    <TaskForm
      onSubmit={task => {
        props.dispatch(addNewTask(task));
        props.history.push("/tasks");
      }}
    />
  );
};

export default connect(null)(AddTask);
