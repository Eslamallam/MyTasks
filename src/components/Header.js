import React from "react";
import { NavLink } from "react-router-dom";
import "../styles/header.css";

const Header = () => {
  return (
    <header>
      <NavLink className="nav-item" to="/home" activeClassName="is-active">
        Home
      </NavLink>
      <NavLink className="nav-item" to="/tasks" activeClassName="is-active">
        Tasks
      </NavLink>
      <NavLink className="nav-item" to="/profile" activeClassName="is-active">
        Profile
      </NavLink>
    </header>
  );
};

export default Header;
