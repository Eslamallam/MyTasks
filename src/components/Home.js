import React from "react";
import "../styles/home.css";

const Home = () => {
  return (
    <div className="home">
      <h1>My Tasks</h1>
      <p>
        This is a simple application to manage daily tasks easily, you can
        create new tasks with category or remove existing one.{" "}
      </p>
    </div>
  );
};

export default Home;
