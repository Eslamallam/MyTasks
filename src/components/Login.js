import React from "react";
import { connect } from "react-redux";
import { login, getUserInfo } from "../actions/Auth";
import axios from "axios";
import "../styles/login.css";

class Login extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      username: "",
      password: "",
      AuthUsername: "",
      AuthUserPassword: "",
      error: ""
    };
  }

  componentDidMount() {
    axios.get("./UserCredentials.json").then(res => {
      this.setState(() => ({
        AuthUsername: res.data[0].username,
        AuthUserPassword: res.data[0].password
      }));
    });
  }

  onUsernameChange = e => {
    const username = e.target.value;
    this.setState(() => ({ username, error: "" }));
  };

  onPasswordChange = e => {
    const password = e.target.value;
    this.setState(() => ({ password, error: "" }));
  };

  onSubmit = e => {
    e.preventDefault();

    const { username, password, AuthUserPassword, AuthUsername } = this.state;

    if (!username || !password) {
      this.setState(() => ({
        error: "Please enter your credentials to login"
      }));
    } else if (username !== AuthUsername || password !== AuthUserPassword) {
      this.setState(() => ({
        error: "Invalid credentials"
      }));
    } else {
      this.props.dispatch(getUserInfo());
      this.props.dispatch(login());
      this.props.history.push("/profile");
    }
  };

  render() {
    return (
      <div className="box-layout">
        {this.state.error && <div className="error">{this.state.error}</div>}
        <form onSubmit={this.onSubmit}>
          <h1>MyTasks</h1>
          <p>Login to manage your tasks </p>
          <input
            type="text"
            placeholder="Enter a username"
            value={this.state.username}
            onChange={this.onUsernameChange}
          />
          <br />
          <input
            type="password"
            placeholder="Enter a password"
            value={this.state.password}
            onChange={this.onPasswordChange}
          />
          <br />
          <button type="submit">Login</button>
        </form>
      </div>
    );
  }
}

const mapStateToProps = state => ({
  credentials: state.auth
});

export default connect(mapStateToProps)(Login);
