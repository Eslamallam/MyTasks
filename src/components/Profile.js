import React from "react";
import { connect } from "react-redux";
import { logout } from "../actions/Auth";
import "../styles/profile.css";

export const Profile = props => {
  const { name, email } = props.userInfo;

  return (
    <div className="profile-box">
      <h1>Welcome</h1>
      <h4>Don't Share your data with anyone</h4>
      <p>Name: {name}</p>
      <p>E-mail: {email}</p>
      <button
        onClick={() => {
          props.dispatch(logout());
        }}
      >
        Logout
      </button>
    </div>
  );
};

const mapStateToProps = state => ({
  userInfo: state.auth
});

export default connect(mapStateToProps)(Profile);
