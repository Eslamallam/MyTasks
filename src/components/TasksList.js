import React from "react";
import { NavLink } from "react-router-dom";
import { connect } from "react-redux";
import TaskItem from "./TaskItem";
import "../styles/tasks.css";

export const TasksList = props => {
  return (
    <div className="task-list">
      <NavLink className="nav-item" to="/createtask">
        Create New
      </NavLink>
      {props.tasks.length > 0 ? (
        props.tasks.map(task => {
          return <TaskItem key={task.id} {...task} />;
        })
      ) : (
        <p>"There is no tasks to show"</p>
      )}
    </div>
  );
};

const mapStateToProps = state => {
  return {
    tasks: state.tasks
  };
};

export default connect(mapStateToProps)(TasksList);
