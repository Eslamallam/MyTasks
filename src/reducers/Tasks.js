import Data from "../Data/Tasks.json";

const tasksDefaultState = Data;

export default (state = tasksDefaultState, action) => {
  switch (action.type) {
    case "ADD_TASK":
      return [...state, action.task];
    case "REMOVE_TASK":
      return state.filter(({ id }) => id !== action.id);

    default:
      return state;
  }
};
