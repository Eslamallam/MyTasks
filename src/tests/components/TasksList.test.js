import React from "react";
import Adapter from "enzyme-adapter-react-16";
import { shallow, configure } from "enzyme";
import toJSON from "enzyme-to-json";
import { TasksList } from "../../components/TasksList";
import Tasks from "../fixtures/tasks";

configure({ adapter: new Adapter() });

test("should render task list with tasks", () => {
  const wrapper = shallow(<TasksList tasks={Tasks} />);
  expect(toJSON(wrapper)).toMatchSnapshot();
});
