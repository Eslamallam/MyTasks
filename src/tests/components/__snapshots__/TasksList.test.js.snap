// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render task list with tasks 1`] = `
<div
  className="task-list"
>
  <NavLink
    className="nav-item"
    to="/createtask"
  >
    Create New
  </NavLink>
  <Connect(TaskItem)
    category="Sport"
    description="Run for 10 min"
    id={1}
    key="1"
  />
  <Connect(TaskItem)
    category="Entertainment"
    description="Watch a movie"
    id={2}
    key="2"
  />
  <Connect(TaskItem)
    category="Work"
    description="Do my tasks"
    id={3}
    key="3"
  />
</div>
`;
